### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by Likh-Alex**
- Implemented all `Dao` interfaces except `GenericDao`
- Modified all `Model` classes to meet the tests requirements

### CODE REVIEW

**Review by ivlytvynets — APPROVED** [review]

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/dao/impl/CommentDaoImpl.java [hunk 1]
```diff
@@ -12,21 +15,63 @@ public CommentDaoImpl(SessionFactory sessionFactory) {
...
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(entity);
+            transaction.commit();
+            return entity;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can not create comment: " + entity.getId(), e);
```
[the remark this answers is not in the document]
Comment (Likh-Alex) [reply src/main/java/core/basesyntax/dao/impl/CommentDaoImpl.java:30]: replaced solely with `entity`

File: src/main/java/core/basesyntax/dao/impl/CommentDaoImpl.java [hunk 2]
```diff
@@ -12,21 +15,63 @@ public CommentDaoImpl(SessionFactory sessionFactory) {
...
+        } finally {
+            if (transaction != null) {
```
[the remark this answers is not in the document]
Comment (Likh-Alex) [reply src/main/java/core/basesyntax/dao/impl/CommentDaoImpl.java:32]: Shame on me, copy paste curse follows me. Fixed everywhere where applicable

File: src/main/java/core/basesyntax/dao/impl/MessageDetailsDaoImpl.java [hunk 3]
```diff
@@ -11,11 +13,32 @@ public MessageDetailsDaoImpl(SessionFactory sessionFactory) {
...
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(entity);
+            transaction.commit();
+            return entity;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can not create MessageDetails: " + entity.toString());
```
[the remark this answers is not in the document]
Comment (Likh-Alex) [reply src/main/java/core/basesyntax/dao/impl/MessageDetailsDaoImpl.java:28]: Understood, thank you. Done as suggested

File: src/main/java/core/basesyntax/dao/impl/SmileDaoImpl.java [hunk 4]
```diff
@@ -12,16 +15,42 @@ public SmileDaoImpl(SessionFactory sessionFactory) {
...
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(entity);
+            transaction.commit();
+            return entity;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can not create comment: " + entity.toString(), e);
```
[the remark this answers is not in the document]
Comment (Likh-Alex) [reply src/main/java/core/basesyntax/dao/impl/SmileDaoImpl.java:30]: Done, thank you

File: src/main/java/core/basesyntax/dao/impl/SmileDaoImpl.java [hunk 5]
```diff
@@ -12,16 +15,42 @@ public SmileDaoImpl(SessionFactory sessionFactory) {
...
+        } finally {
+            if (transaction != null) {
```
[the remark this answers is not in the document]
Comment (Likh-Alex) [reply src/main/java/core/basesyntax/dao/impl/SmileDaoImpl.java:32]: Done, thank you

**Review by Sofasmile — APPROVED** [review]

**Pull request closed without merging**